File: skills/desktop-app-dev/clients/MediaClient.java
import com.fasterxml.jackson.databind.ObjectMapper;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.util.Map;

public final class MediaClient {
    private final HttpClient http;
    private final String baseUrl;
    private final String token;
    private final ObjectMapper mapper = new ObjectMapper();

    public MediaClient(String baseUrl, String token) {
        this.http = HttpClient.newHttpClient();
        this.baseUrl = baseUrl;
        this.token = token;
    }

    private HttpRequest.Builder auth(HttpRequest.Builder builder) {
        if (token != null) {
            builder.header("Authorization", "Bearer " + token);
        }
        return builder;
    }

    public String enqueue(String kind, Map<String, Object> payload, String dedupeKey)
            throws Exception {
        String body = mapper.writeValueAsString(
                Map.of("kind", kind, "payload", payload, "dedupe_key", dedupeKey));
        HttpRequest request = auth(HttpRequest.newBuilder())
                .uri(URI.create(baseUrl + "/tasks"))
                .header("Content-Type", "application/json")
                .POST(HttpRequest.BodyPublishers.ofString(body))
                .build();
        return http.send(request, HttpResponse.BodyHandlers.ofString()).body();
    }

    public String task(long id) throws Exception {
        HttpRequest request = auth(HttpRequest.newBuilder())
                .uri(URI.create(baseUrl + "/tasks/" + id))
                .GET()
                .build();
        return http.send(request, HttpResponse.BodyHandlers.ofString()).body();
    }

    public String depsProgress() throws Exception {
        HttpRequest request = auth(HttpRequest.newBuilder())
                .uri(URI.create(baseUrl + "/deps/progress"))
                .GET()
                .build();
        return http.send(request, HttpResponse.BodyHandlers.ofString()).body();
    }

    public String depsStatus() throws Exception {
        HttpRequest request = auth(HttpRequest.newBuilder())
                .uri(URI.create(baseUrl + "/deps/status"))
                .GET()
                .build();
        return http.send(request, HttpResponse.BodyHandlers.ofString()).body();
    }

    public String installDeps() throws Exception {
        HttpRequest request = auth(HttpRequest.newBuilder())
                .uri(URI.create(baseUrl + "/deps/install"))
                .header("Content-Type", "application/json")
                .POST(HttpRequest.BodyPublishers.ofString("{}"))
                .build();
        return http.send(request, HttpResponse.BodyHandlers.ofString()).body();
    }
}

File: skills/desktop-app-dev/clients/media_client.cpp
#include <string>
#include <utility>

#include <curl/curl.h>

class MediaClient {
public:
    explicit MediaClient(
        std::string baseUrl = "http://127.0.0.1:8765",
        std::string token = "")
        : baseUrl_(std::move(baseUrl)), token_(std::move(token)) {}

    std::string enqueue(const std::string& jsonBody) const {
        return post("/tasks", jsonBody);
    }

    std::string task(long id) const {
        return get("/tasks/" + std::to_string(id));
    }

    std::string depsProgress() const {
        return get("/deps/progress");
    }

    std::string depsStatus() const {
        return get("/deps/status");
    }

    std::string installDeps() const {
        return post("/deps/install", "{}");
    }

private:
    std::string baseUrl_;
    std::string token_;

    static size_t writeCallback(void* contents, size_t size, size_t nmemb, void* userp) {
        auto* output = static_cast<std::string*>(userp);
        output->append(static_cast<char*>(contents), size * nmemb);
        return size * nmemb;
    }

    std::string request(const std::string& path, const std::string& body, bool isPost) const {
        std::string url = baseUrl_ + path;
        std::string result;
        CURL* curl = curl_easy_init();
        if (!curl) {
            return result;
        }
        curl_easy_setopt(curl, CURLOPT_URL, url.c_str());
        curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, writeCallback);
        curl_easy_setopt(curl, CURLOPT_WRITEDATA, &result);
        curl_slist* headers = nullptr;
        if (!token_.empty()) {
            std::string auth = "Authorization: Bearer " + token_;
            headers = curl_slist_append(headers, auth.c_str());
            curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
        }
        if (isPost) {
            curl_easy_setopt(curl, CURLOPT_POST, 1L);
            curl_easy_setopt(curl, CURLOPT_POSTFIELDS, body.c_str());
        }
        curl_easy_perform(curl);
        if (headers) {
            curl_slist_free_all(headers);
        }
        curl_easy_cleanup(curl);
        return result;
    }

    std::string get(const std::string& path) const {
        return request(path, "", false);
    }

    std::string post(const std::string& path, const std::string& body) const {
        return request(path, body, true);
    }
};
